<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<html>
 <head>
  <meta http-equiv="content-type" content="text/html; charset=UTF-8">
  <title>BadFunctionCallException クラス</title>
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <link media="all" rel="stylesheet" type="text/css" href="style.css"/>
 </head>
 <body class="docs"><div class="manualnavbar" style="text-align: center;">
 <div class="prev" style="text-align: left; float: left;"><a href="spl.exceptions.html">例外</a></div>
 <div class="next" style="text-align: right; float: right;"><a href="class.badmethodcallexception.html">BadMethodCallException</a></div>
 <div class="up"><a href="spl.exceptions.html">例外</a></div>
 <div class="home"><a href="index.html">PHP Manual</a></div>
</div><hr /><div id="class.badfunctioncallexception" class="reference">
 <h1 class="title">BadFunctionCallException クラス</h1>
 
 
 <div class="partintro"><p class="verinfo">(PHP 5 &gt;= 5.1.0)</p>
 

  <div class="section" id="badfunctioncallexception.intro">
   <h2 class="title">導入</h2>
   <p class="para">
    未定義の関数をコールバックが参照したり、引数を指定しなかったりした場合にスローされる例外です。
   </p>
  </div>

 
  <div class="section" id="badfunctioncallexception.synopsis">
   <h2 class="title">クラス概要</h2>
 

   <div class="classsynopsis">
    <div class="ooclass"></div>
 

    <div class="classsynopsisinfo">
     <span class="ooclass">
      <strong class="classname">BadFunctionCallException</strong>
     </span>
 
    <span class="ooclass">
      <span class="modifier">extends</span>
      <a href="class.logicexception.html" class="classname">LogicException</a>
     </span>
     {</div>

 
    <div class="classsynopsisinfo classsynopsisinfo_comment">/* 継承したプロパティ */</div>
    <div class="fieldsynopsis">
     <span class="modifier">protected</span>
     <span class="type">string</span>
      <var class="varname"><a href="class.exception.html#exception.props.message">$<var class="varname">message</var></a></var>
    ;</div>
<div class="fieldsynopsis">
     <span class="modifier">protected</span>
     <span class="type">int</span>
      <var class="varname"><a href="class.exception.html#exception.props.code">$<var class="varname">code</var></a></var>
    ;</div>
<div class="fieldsynopsis">
     <span class="modifier">protected</span>
     <span class="type">string</span>
      <var class="varname"><a href="class.exception.html#exception.props.file">$<var class="varname">file</var></a></var>
    ;</div>
<div class="fieldsynopsis">
     <span class="modifier">protected</span>
     <span class="type">int</span>
      <var class="varname"><a href="class.exception.html#exception.props.line">$<var class="varname">line</var></a></var>
    ;</div>



   <div class="classsynopsisinfo classsynopsisinfo_comment">/* 継承したメソッド */</div>
    <div class="methodsynopsis dc-description">
   <span class="modifier">final</span> <span class="modifier">public</span> <span class="type">string</span> <span class="methodname"><a href="exception.getmessage.html" class="methodname">Exception::getMessage</a></span>
    ( <span class="methodparam">void</span>
   )</div>
<div class="methodsynopsis dc-description">
   <span class="modifier">final</span> <span class="modifier">public</span> <span class="type">Exception</span> <span class="methodname"><a href="exception.getprevious.html" class="methodname">Exception::getPrevious</a></span>
    ( <span class="methodparam">void</span>
   )</div>
<div class="methodsynopsis dc-description">
   <span class="modifier">final</span> <span class="modifier">public</span> <span class="type">mixed</span> <span class="methodname"><a href="exception.getcode.html" class="methodname">Exception::getCode</a></span>
    ( <span class="methodparam">void</span>
   )</div>
<div class="methodsynopsis dc-description">
   <span class="modifier">final</span> <span class="modifier">public</span> <span class="type">string</span> <span class="methodname"><a href="exception.getfile.html" class="methodname">Exception::getFile</a></span>
    ( <span class="methodparam">void</span>
   )</div>
<div class="methodsynopsis dc-description">
   <span class="modifier">final</span> <span class="modifier">public</span> <span class="type">int</span> <span class="methodname"><a href="exception.getline.html" class="methodname">Exception::getLine</a></span>
    ( <span class="methodparam">void</span>
   )</div>
<div class="methodsynopsis dc-description">
   <span class="modifier">final</span> <span class="modifier">public</span> <span class="type">array</span> <span class="methodname"><a href="exception.gettrace.html" class="methodname">Exception::getTrace</a></span>
    ( <span class="methodparam">void</span>
   )</div>
<div class="methodsynopsis dc-description">
   <span class="modifier">final</span> <span class="modifier">public</span> <span class="type">string</span> <span class="methodname"><a href="exception.gettraceasstring.html" class="methodname">Exception::getTraceAsString</a></span>
    ( <span class="methodparam">void</span>
   )</div>
<div class="methodsynopsis dc-description">
   <span class="modifier">public</span> <span class="type">string</span>  <span class="methodname"><a href="exception.tostring.html" class="methodname">Exception::__toString</a></span>
    ( <span class="methodparam">void</span>
   )</div>
<div class="methodsynopsis dc-description">
   <span class="modifier">final</span> <span class="modifier">private</span> <span class="type">void</span> <span class="methodname"><a href="exception.clone.html" class="methodname">Exception::__clone</a></span>
    ( <span class="methodparam">void</span>
   )</div>

   }</div>

 
  </div>
 
 </div>
 
 
 
</div>
<div id="usernotes">
 <div class="head">
  <h3 class="title">User Contributed Notes</h3>
 </div>
<div id="allnotes">  <a name="102229""></a>
  <div class="note">
   <strong class="user">Dawid Krysiak</strong>
   <a href="#102229" class="date">03-Feb-2011 11:35</a>
   <div class="text">
    <div class="phpcode">
<code><span class="html">
Direct known subclasses:<br />
BadMethodCallException<br />
<br />
So BadMethodCallException can be used in case the subject function is or should be a part of class / object - it's use defines the case more precisely.</span>
</code>
    </div>
   </div>
  </div>
 <div class="foot"></div>
  <a name="94635""></a>
  <div class="note">
   <strong class="user">tom at tomwardrop dot com</strong>
   <a href="#94635" class="date">16-Nov-2009 04:42</a>
   <div class="text">
    <div class="phpcode">
<code><span class="html">
A typical use for this exception, is in conjunction with the is_callable() function.</span>
</code>
    </div>
   </div>
  </div>
 <div class="foot"></div>
</div>
</div>
<hr /><div class="manualnavbar" style="text-align: center;">
 <div class="prev" style="text-align: left; float: left;"><a href="spl.exceptions.html">例外</a></div>
 <div class="next" style="text-align: right; float: right;"><a href="class.badmethodcallexception.html">BadMethodCallException</a></div>
 <div class="up"><a href="spl.exceptions.html">例外</a></div>
 <div class="home"><a href="index.html">PHP Manual</a></div>
</div></body></html>
